### PR TITLE
:arrow_up: [#50] updated dependecies and added oidc tests

### PR DESCRIPTION
**Pull request by bart-maykin**
Fixes [#50](https://github.com/maykinmedia/open-api-framework/issues/50)

**Changes**

updated libraries:
- billiard `3.6.4.0` to `4.2.0`
- celery `5.2.7` to `5.4.0`
- certifi `2020.07.4` to `2023.7.04`
- django `4.2.11` to `4.2.14`
- django-axes `6.4.0` to `6.5.1`
- django-cors-headers `4.3.1` to `4.4.0`
- django-filters `23.2` to `24.2`
- django-jsonform `2.21.4` to `2.22.0`
- djangorestframework `3.14.0` to `3.15.2`
- maykin-2fa `1.0.0` to `1.0.1`
- mozilla-django-oidc-db `0.16.0` to `0.19.0`
- open-api-framework `0.5.0` to `0.6.1`
- requests `2.31.0` to `2.32.3`
- sentry-sdk `1.45.0` to `2.12.0`
- tornado `6.4` to `6.4.1`
- urllib3 `2.2.1` to `2.2.2`
- vine `5.0.0` to `5.1.0`

updated oidc tests to work for new library version

### CODE REVIEW

**Review by annashamray — APPROVED** [review]
Approved in relation to updating dependencies
@stevenbal could you please review OIDC part?

File: package.json [hunk 1]
```diff
@@ -20,6 +20,7 @@
   "license": "EUPL",
   "homepage": "https://maykinmedia.nl",
   "dependencies": {
+    "fix": "^0.0.6",
```
Comment (stevenbal) [inline package.json:23]: What do we need this for?

File: src/objecttypes/accounts/tests/factories.py [hunk 2]
```diff
@@ -1,11 +1,29 @@
...
+
+    class Params:
+        superuser = factory.Trait(
+            is_staff=True,
+            is_superuser=True,
+        )
+
+
+class StaffUserFactory(UserFactory):
+    is_staff = True
```
Comment (stevenbal) [inline src/objecttypes/accounts/tests/factories.py:26]: What is the difference between these? Doesn't `UserFactory` now also have `is_staff=True?`

File: src/objecttypes/utils/tests/keycloak.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+from django.apps import apps
+
+from pyquery import PyQuery as pq
+from requests import Session
+
+KEYCLOAK_BASE_URL = "http://localhost:8080/realms/test/protocol/openid-connect"
+
+
+def keycloak_login(
```
Comment (stevenbal) [inline src/objecttypes/utils/tests/keycloak.py:12]: Can't we reuse this from `mozilla-django-oidc-db`? https://github.com/maykinmedia/mozilla-django-oidc-db/blob/d0630df6a5dc9d8bf34dbed5a0be983973779ed6/tests/utils.py#L7

**Review by stevenbal — CHANGES_REQUESTED** [review]
Some minor comments/questions. It also seems that CI is failing? Not sure if it's related https://github.com/maykinmedia/objecttypes-api/actions/runs/10321355943/job/28575706632?pr=122

File: src/objecttypes/utils/tests/keycloak.py (hunk 3 above)
Comment (bart-maykin) [reply src/objecttypes/utils/tests/keycloak.py:12]: This is reused from mozilla-django-oidc-db but with some slight alterations. Just as in open-forms.

Screenshot from 2024-08-14 16-59-11

**Review by stevenbal — APPROVED** [review]

**Pull request merged**